### PR TITLE
[cli] [vtgate/vschemaacl] Migrate VschemaACL flags to pflags

### PR DESCRIPTION
**Pull request by rsajwani**
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description
Updates VschemaACL related flag references specified in https://github.com/vitessio/vitess/issues/10936

go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ "vitess.io/vitess/go/vt/vtgate/vschemaacl" {print $1}'                                                                                                      
vitess.io/vitess/go/cmd/vtadmin
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtexplain
vitess.io/vitess/go/cmd/vtgate
vitess.io/vitess/go/cmd/vtgateclienttest

## Related Issue(s)

#10936

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by ajm188 — APPROVED** [review]
LGTM, but the title/description has a typo:

> VsSchemaACL

should be `vschemaACL`

File: go/cmd/vtcombo/main.go [hunk 1]
```diff
@@ -78,6 +80,7 @@ func init() {
 	servenv.RegisterGRPCServerFlags()
 	servenv.RegisterGRPCServerAuthFlags()
 	servenv.RegisterServiceMapFlag()
+	servenv.OnParseFor("vtcombo", vschemaacl.RegisterSchemaACLFlags)
```
Comment (ajm188) [inline go/cmd/vtcombo/main.go:83]: why are these not in `package vschemaacl`? (same comment applies to the identical line in `go/cmd/vtgate/`

File: go/cmd/vtcombo/main.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ import (
 	"strings"
 	"time"
 
+	"vitess.io/vitess/go/vt/vtgate/vschemaacl"
```
Comment (ajm188) [inline go/cmd/vtcombo/main.go:30]: below the `GitHub.com/**` imports please :)

File: go/vt/vtgate/vschemaacl/vschemaacl.go [hunk 3]
```diff
@@ -38,17 +35,17 @@ var (
...
 	fs.StringVar(&AuthorizedDDLUsers, "vschema_ddl_authorized_users", AuthorizedDDLUsers, "List of users authorized to execute vschema ddl operations, or '%' to allow all users.")
 }
 
 // Init parses the users option and sets allowAll / acl accordingly
 func Init() {
-	// register flag
-	for _, cmd := range []string{"vtadmin", "vtcombo", "vtgate", "vtgateclienttest"} {
-		servenv.OnParseFor(cmd, registerSchemaACLFlags)
-	}
-
```
Comment (ajm188) [inline go/vt/vtgate/vschemaacl/vschemaacl.go]: why did you remove this?

**Review by ajm188 — COMMENTED** [review]
i'm confused by the latest round of changes, i think we want back (most) of the registration code in `vchemaacl/vschemaacl.go`

File: go/cmd/vtcombo/main.go (hunk 1 above)
Comment (rsajwani) [reply go/cmd/vtcombo/main.go]: So I made mistake. I have reverted back my change. Vschemaacl has Init() method which I mixed up with init(). I put registration in Init() and expected the flags to be available for binaries but it was not happening thats why I moved them under each binary endpoint. But now I realized it was because of Init() is not "init()" :D

File: go/vt/vtgate/vschemaacl/vschemaacl.go (hunk 3 above)
Comment (rsajwani) [reply go/vt/vtgate/vschemaacl/vschemaacl.go]: So I made mistake. I have reverted back my change. Vschemaacl has Init() method which I mixed up with init(). I put registration in Init() and expected the flags to be available for binaries but it was not happening thats why I moved them under each binary endpoint. But now I realized it was because of Init() is not "init()" :D

File: go/test/endtoend/backup/transform/backup_transform_utils.go [hunk 4]
```diff
@@ -368,7 +368,8 @@ func verifySemiSyncStatus(t *testing.T, vttablet *cluster.Vttablet, expectedStat
 // verifyInitialReplication generates a record on the primary and verifies that the record
 // exists on all tablets in the shard. We also check the PRIMARY as with lossless semi-sync
 // it's possible that a GTID is applied on a replica but not (yet) on the source:
-//   http://my-replication-life.blogspot.com/2013/09/loss-less-semi-synchronous-replication.html
+//
```
Comment (rsajwani) [inline go/test/endtoend/backup/transform/backup_transform_utils.go:371]: This looks like unnecessary space. I can fix it in my subsequent PR.

File: go/vt/vtgate/vschemaacl/vschemaacl.go [hunk 5]
```diff
@@ -34,19 +37,35 @@ var (
...
+//
+// `go/cmd/*` entrypoints should either use servenv.ParseFlags(WithArgs)? which
+// calls this function, or call this function directly before parsing
+// command-line arguments.
+func RegisterSchemaACLFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&AuthorizedDDLUsers, "vschema_ddl_authorized_users", AuthorizedDDLUsers, "List of users authorized to execute vschema ddl operations, or '%' to allow all users.")
+}
+
+func init() {
+	// register flag
```
Comment (ajm188) [inline go/vt/vtgate/vschemaacl/vschemaacl.go:50]: delete this comment

File: go/vt/vtgate/vschemaacl/vschemaacl.go [hunk 6]
```diff
@@ -34,19 +37,35 @@ var (
...
+	for _, cmd := range []string{"vtexplain", "vtcombo", "vtgate", "vtgateclienttest"} {
```
Comment (ajm188) [inline go/vt/vtgate/vschemaacl/vschemaacl.go:51]: on second thought ... are we sure that `vtexplain` actually needs ACLs? i had the understanding it was entirely a local simulation, so you shouldn't need to be "authorized"

Comment (rsajwani) [reply go/vt/vtgate/vschemaacl/vschemaacl.go:51]: I am not sure ... I added it because vtexplain help file has it so I added it to the array of cmd
